woocommerce/woocommerce
# Fix order action aria-labels to be unique

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Updates the aria labels introduced to the order actions screen in #49424 to ensure they are unique.

Currently each button has the same label `View order %s`, this will cause confusion to users of assistive technology as the buttons are identical to screen-readers making use of the label.

The changes to this pull request are two-fold:

* In the My Account > Orders template, a default label is used in the form `[Button Name] order [order number]`
* In `wc_get_account_orders_actions()` `aria-label` properties are introduced to allow for developers to customise the label as needed.
   * In WooCommerce Core, this is to allow the aria label for `pay` to be the more grammatically correct `pay for order xx` rather than `pay order xx`. 
   * In extensions, this will be particularly helpful for avoiding phrases such as `cancel pre-order order xx`

Closes #51404 .



### How to test the changes in this Pull Request:

The purpose is to test the aria labels on the buttons in My Account > Orders

1. Create a new simple product (if required)
2. Purchase the product 
3. In the admin, update the new order's status to `Pending Payment` (this will ensure all actions are shown).
4. In the store, navigate to My Account > Orders
5. Enable your screen reader
6. Hover over/focus on the pay button, ensure the action is announced as "Pay for order [order number]"
7. Hover over/focus on the view button, ensure the action is announced as "View order [order number]"
8. Hover over/focus on the cancel button, ensure the action is announced as "Cancel order [order number]"


### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message  

Ensure the order action aria-labels are different for each action.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
